azure-pipelines.yml

```diff
@@ -157,15 +206,31 @@ steps:
     displayName: Cache Maven local repo and wrapper
     condition: ne(variables.SKIP_CACHE, 'true')
 
+  # Main build and test execution
   - bash: |
       set -e
       ./mvnw --version
-      echo "ON_CRON_ONLY:"$ON_CRON_ONLY
-      echo "BUILD_REASON:"$BUILD_REASON
-      echo "cmd: "$(cmd)
-      eval "$(cmd)"
+      echo "ON_CRON_ONLY: $ON_CRON_ONLY"
+      echo "BUILD_REASON: $BUILD_REASON"
+      echo "REQUIRE_DISPLAY: $REQUIRE_DISPLAY"
+      echo "AGENT_OS: $AGENT_OS"
+      echo "cmd: $(cmd)"
+      # Conditionally run with xvfb for Linux GUI tests
+      if [ "$REQUIRE_DISPLAY" = "true" ] && [ "$AGENT_OS" = "Linux" ]; then
+        echo "Starting xvfb for GUI tests"
+        export DISPLAY=:99
+        Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
```

Comment:
Please share a root reason failure that forcing you to go this route.

Reply:
This can mean that the same failure can be on users locals.
If that is potentially possible, I would rather exclude tests for this UI classes.